aimmo-game/tests/functional/test_map_movements.py

```diff
@@ -0,0 +1,195 @@
+from unittest import TestCase
+
+from mock_world import MockWorld
+from simulation.location import Location
+from simulation import map_generator
+from tests.test_simulation.dummy_avatar import MoveEastDummy, MoveWestDummy, MoveNorthDummy, MoveSouthDummy, WaitDummy
+
+
+class TestMapMovements(TestCase):
+
+    SETTINGS = {
+        'START_HEIGHT': 50,
+        'START_WIDTH': 50,
+        'OBSTACLE_RATIO': 0,
+    }
+
+    def set_up_environment(self, dummy_list=None, location=Location(0, 0), map_generator_class=map_generator.Main):
+        """
+        Utility method for testing.
+        """
+        self.game = MockWorld(TestMapMovements.SETTINGS, dummy_list, map_generator_class)
+        self.game.game_state.add_avatar(1, None, location)
+        self.avatar = self.game.avatar_manager.get_avatar(1)
+
+    def test_movement_five_times_in_all_directions(self):
```

Comment:
can't all these bits be more factorised somehow? Seems  a bit clunky. Also when you write "consists of 5 tests" or "4 tests", you sort of expect them to be in different functions?

Reply:
I used to put stuff like this in separate tests but in a previous PR was told by Dips and Duncan not to do so if I'm only changing certain values but not the actual edge case. I think here we're testing only one edge case in 4 different scenario's which is why I left it inside one test.

I will change the comment though! :+1: